docs/performance/010.md

```diff
@@ -0,0 +1,27 @@
+---
+title: CombinePipedModuleFunctionsAnalyzer
+category: performance
+categoryindex: 4
+index: 3
+---
+
+# CombinePipedModuleFunctionsAnalyzer
+
+## Problem
+
+Subsequent collection function like `List.filter` and `List.map` can sometimes be combined into `List.choose`.
```

Comment:
```suggestion
Subsequent collection functions like `List.filter` and `List.map` can sometimes be combined into `List.choose`.
```

Reply:
Besides of List.choose, there are a few other map+reduce functions that can combine other list-operations to avoid iterating lists multiple times through, e.g. instead of doing list-operations and then List.concat in the end, it can sometimes be replaced with List.collect. And sometimes List.map + List.sum can be replaced with List.sumBy.

Reply:
Thanks, good input. I'm going to consider this out of scope for this PR, feel free to create an extension of this PR once it gets merged.